weaviate/collection/extract_filters.py

```
@@ -99,10 +99,70 @@ def __filter_to_int_list(value: FilterValues) -> Optional[weaviate_pb2.IntArray]
+
+class FilterToREST:
+    @staticmethod
+    def convert(weav_filter: Optional[_Filters]) -> Optional[Dict[str, Any]]:
+        if weav_filter is None:
+            return None
+        if isinstance(weav_filter, _FilterValue):
+            return FilterToREST.__value_filter(weav_filter)
+        else:
+            return FilterToREST.__and_or_filter(weav_filter)
+
+    @staticmethod
+    def __value_filter(weav_filter: _FilterValue) -> Dict[str, Any]:
+        return {
+            "operator": weav_filter.operator.value,
+            "path": weav_filter.path if isinstance(weav_filter.path, list) else [weav_filter.path],
+            **FilterToREST.__parse_filter(weav_filter.value),
+        }
+
+    @staticmethod
+    def __parse_filter(value: FilterValues) -> Dict[str, Any]:
+        if isinstance(value, str):
+            return {"valueText": value}
+        if isinstance(value, uuid_lib.UUID):
+            return {"valueText": str(value)}
+        if isinstance(value, TIME):
+            return {"valueDate": _datetime_to_string(value)}
+        if isinstance(value, bool):
+            return {"valueBoolean": value}
+        if isinstance(value, int):
+            return {"valueInt": value}
+        if isinstance(value, float):
+            return {"valueNumber": value}
+        if isinstance(value, list):
+            if isinstance(value[0], str):
+                return {"valueTextArray": value}
+            if isinstance(value[0], uuid_lib.UUID):
+                return {"valueTextArray": [str(val) for val in value]}
+            if isinstance(value[0], TIME):
+                return {"valueDateArray": [_datetime_to_string(cast(TIME, val)) for val in value]}
+            if isinstance(value[0], bool):
+                return {"valueBooleanArray": value}
+            if isinstance(value[0], int):
+                return {"valueIntArray": value}
+            if isinstance(value[0], float):
+                return {"valueNumberArray": value}
+        return {}
```

should we have an exception here? Better complain than failing silently?